source/core/GenericFunctions.cpp

```diff
@@ -683,13 +683,18 @@ VIREO_FUNCTION_SIGNATURET(VectorMaxMinOp, AggregateMaxAndMinInstruction)
     IntIndex elementSizeY = srcArrayY->ElementType()->TopAQSize();
     IntIndex elementSizeMax = maxArray->ElementType()->TopAQSize();
     IntIndex elementSizeMin = minArray->ElementType()->TopAQSize();
+    IntIndex* newDimensionLengths = srcArrayX->DimensionLengths();
+    IntIndex rank = srcArrayX->Rank();
     IntIndex count = srcArrayX->Length();
-    if (srcArrayY->Length() < count)
+    if (srcArrayY->Length() < count) {
```

Comment:
I am not sure whether this is right. If the count is same but dimension length is different for the different dimensions, is this doing the right thing (as done by LabVIEW). I think we should write test cases for the different use cases as discussed in person.